src/net.h

```diff
@@ -519,7 +520,7 @@ class CNode
     // BUIP010:
     bool ThinBlockCapable()
     {
-        if (nServices & NODE_XTHIN)
+        if ((nServices & NODE_XTHIN) && (nServices & NODE_BLOOM))
```

Comment:
is this something we should have changed in any case?

Reply:
yes this should have always been like this...it's more stringent...we always need bloom filtering if we're going to use the xthin service.

Reply:
Why do you say that? It's safe to disable "filterload", "filteradd" and "filterclear" without removing xthin support.